build/vs2019/FreeCam.h

```diff
@@ -0,0 +1,63 @@
+#pragma once
+
+
+#include <glm/glm.hpp>
+#include <glm/gtc/matrix_transform.hpp>
+#include <glm/gtx/rotate_vector.hpp>
+#include <GLFW/glfw3.h>
+#include "Engine/Input/InputHandler.h"
+
+namespace gp1 {
+	class FreeCam {
+	public:
+		FreeCam()
+			:ViewMatrix(glm::mat4(1.0f))
+		{
+			input::InputGroup* freecamGroup = input::InputHandler::GetOrCreateInputGroup("freecam");
+
+			freecamGroup->CreateAxisInputBinding("lookUp", 1)->BindCallback(std::bind(&FreeCam::LookUp, this, std::placeholders::_1));
+
+			freecamGroup->CreateAxisInputBinding("lookRight", 0)->BindCallback(std::bind(&FreeCam::LookRight, this, std::placeholders::_1));
+
+			freecamGroup->CreateButtonInputBinding("forward", GLFW_KEY_W)->BindCallback(std::bind(&FreeCam::MoveForward, this, std::placeholders::_1));
+
+			freecamGroup->CreateButtonInputBinding("backward", GLFW_KEY_S)->BindCallback(std::bind(&FreeCam::MoveBackward, this, std::placeholders::_1));
+
+			freecamGroup->CreateButtonInputBinding("left", GLFW_KEY_A)->BindCallback(std::bind(&FreeCam::MoveLeft, this, std::placeholders::_1));
+
+			freecamGroup->CreateButtonInputBinding("right", GLFW_KEY_D)->BindCallback(std::bind(&FreeCam::MoveRight, this, std::placeholders::_1));
+		}
```

Comment:
Avoid implementations in header files, unless they are one-liners and single-instruction 😊